pyxem/utils/segment_utils.py

```
@@ -0,0 +1,313 @@
+        are discarded.
+    max_size : float
+        Grains with size (i.e. total number of pixels) above max_size
+        are discarded.
+    max_number_of_grains : int
+        Maximum number of grains included in the returned separated
+        grains. If it is exceeded, those with highest peak intensities
+        will be returned.
+    marker_radius : float
+        If 1 or larger, each marker for watershed is expanded to a disk
+        of radius marker_radius. marker_radius should not exceed
+        2*min_distance.
+    threshold : bool
+        If True, a mask is calculated by thresholding the VDF image by
+        the Li threshold method in scikit-image. If False (default), the
+        mask is the boolean VDF image.
+    exclude_border : int or True, optional
+        If non-zero integer, peaks within a distance of exclude_border
+        from the boarder will be discarded. If True, peaks at or closer
+        than min_distance of the boarder, will be discarded.
+    plot_on : bool
+        If True, the VDF, the mask, the distance transform
+        and the separated grains will be plotted in one figure window.
+
+    Returns
+    -------
+    sep : np.array
+        Array containing segments from VDF images (i.e. separated
+        grains). Shape: (image size x, image size y, number of grains)
+
+    References
+    ----------
+    [1] http://scikit-image.org/docs/dev/auto_examples/segmentation/
+        plot_watershed.html
+    [2] http://scikit-image.org/docs/dev/auto_examples/xx_applications/
+        plot_coins_segmentation.html#sphx-glr-auto-examples-xx-
+        applications-plot-coins-segmentation-py
+    """
+
+    # Create a mask from the input VDF image.
+    if threshold:
+        th = threshold_li(vdf_temp)
+        if np.isnan(th):
+            th = 0.
+        mask = np.zeros_like(vdf_temp)
+        mask[vdf_temp > th] = True
+    else:
+        mask = vdf_temp.astype('bool')
+    if np.any(np.nonzero(mask)) is False:
```

What is the intention with this if statement? It looks a little odd to force a return None - wouldn't it be better to raise an error? In any case this condition needs a test.

---

Will remove this checking. 

You've got to have put garbage in to be triggering this in, and as the saying goes, Garbage In Garbage Out.